AppCenterPush/AppCenterPush/MSPush.m

```diff
@@ -219,16 +219,7 @@ - (void)registerForRemoteNotifications {
 #if TARGET_OS_OSX
   [NSApp registerForRemoteNotificationTypes:(NSRemoteNotificationTypeSound | NSRemoteNotificationTypeBadge)];
 #elif TARGET_OS_IOS
-  if (floor(NSFoundationVersionNumber) <= NSFoundationVersionNumber_iOS_9_x_Max) {
-    UIUserNotificationType allNotificationTypes =
-        (UIUserNotificationType)(UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
-    UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:allNotificationTypes categories:nil];
-    [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
-  } else {
-
-// Ignore the partial availability warning as the compiler doesn't get that we checked for pre-iOS 10 already.
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wpartial-availability"
+  if (@available(iOS 10.0, *)) {
```

Comment:
Not necessary. It is already in `#elif TARGET_OS_IOS` condition and 10.0 is the minimum version of iOS that App Center supports.

Reply:
Same here.